cassiopeia/preprocess/UMI_utils.py

```diff
@@ -68,6 +69,10 @@
 filter_func = lambda al: al.has_tag(CELL_BC_TAG)
 
 
+class UMIUtilsError(Exception):
+    pass
+
+
```

Comment:
I'd opt for using a general `PreprocessingError` -- thoughts?

Reply:
We do have a `PreprocessError` defined in `preprocess.py`, but we can't import this into `UMI_utils` because it'll cause a circular import. Do you think its fine to re-define a new error with the same name in `UMI_utils.py`?

Reply:
What if we make a new file called `_errors_and_warnings.py` that we can import from? If you think this is stupid, we can just use the `UMIUtilsError` that you defined here.

Reply:
That's a good idea